### PR TITLE
add dumping callstack to kineto

### PR DESCRIPTION
**Pull request by gaoteng-git**
In profiler, pass operators' callstack to kineto and dump them into chrome tracing file.
The kineto side update is merged [here](https://github.com/pytorch/kineto/commit/66a4cad380d3ce4fb8d7b38351f841899d9bf950)

### CODE REVIEW

**Comment by gdankel** [comment]
@gaoteng-git can you please update to latest master?

File: torch/csrc/autograd/profiler_kineto.cpp [hunk 1]
```diff
@@ -211,6 +215,17 @@ std::string shapesToStr(const std::vector<std::vector<int64_t>>& shapes) {
   return oss.str();
 }
 
+std::string stacksToStr(const std::vector<std::string>& stacks) {
+  std::ostringstream oss;
+  for (auto idx = 0; idx < stacks.size(); ++idx) {
```
Comment (ilia-cher) [inline torch/csrc/autograd/profiler_kineto.cpp:220]: please fix clang-tidy here

**Review by ilia-cher — COMMENTED** [review]
LG, just a linter issue

File: torch/csrc/autograd/profiler_kineto.cpp (hunk 1 above)
Comment (gaoteng-git) [reply torch/csrc/autograd/profiler_kineto.cpp:220]: > please fix clang-tidy here

This strictly follows the code style in the above function "shapesToStr". So I wonder if using "auto" is better than the long string "std::vector<std::string>::size_type".

Comment (gaoteng-git) [reply torch/csrc/autograd/profiler_kineto.cpp]: Thanks. Done.

**Comment by gaoteng-git** [comment]
> @gaoteng-git can you please update to latest master?

Thanks. Done.

File: torch/csrc/autograd/profiler_kineto.cpp (hunk 1 above)
Comment (ilia-cher) [reply torch/csrc/autograd/profiler_kineto.cpp]: it can be just `size_t`

File: torch/csrc/autograd/profiler_kineto.cpp [hunk 2]
```diff
@@ -211,6 +215,17 @@ std::string shapesToStr(const std::vector<std::vector<int64_t>>& shapes) {
...
 
+std::string stacksToStr(const std::vector<std::string>& stacks) {
+  std::ostringstream oss;
+  for (std::vector<std::string>::size_type idx = 0; idx < stacks.size(); ++idx) {
+    if (idx > 0) {
+      oss << ";";
+    }
+    oss << stacks[idx];
+  }
+  return oss.str();
```
Comment (malfet) [inline torch/csrc/autograd/profiler_kineto.cpp:226]: Can we do it an a bit more C++ way:
```suggestion
  std::copy(stacks.begin(), stacks.end(), std::ostream_iterator<std::string>(oss, ";"));
  auto rc = oss.str();
  rc.pop_back();
  return rc;
```

Comment (ilia-cher) [reply torch/csrc/autograd/profiler_kineto.cpp:226]: @gaoteng-git could you please update?

**Pull request closed without merging**